In [ ]:
막 위아래 순서 좀 다시 돌려본 코드들 있어서 ... 나중엔 살짝 알아보기 힘들거 같긴한데...

서울시 인구 데이터 불러오기

In [ ]:
import pandas as pd

# CSV 파일 경로 지정
file_path = '/content/201_DT_201003_A010006_20240228235534.csv'

# 특정 행을 제외하고 데이터 불러오기
rows_to_skip = [29832]  # 제외할 행의 번호를 리스트로 지정
df = pd.read_csv(file_path, skiprows=rows_to_skip)

# 불러온 데이터 확인
print(df)

         동별      연령별     항목  단위 2023. 3/4 2023. 4/4  Unnamed: 6
0        합계       합계      계 NaN   9655918   9638799         NaN
1        합계       합계    한국인 NaN   9407540   9386034         NaN
2        합계       합계  등록외국인 NaN    248378    252765         NaN
3        합계     0~4세      계 NaN    220964    218060         NaN
4        합계     0~4세    한국인 NaN    216208    213014         NaN
...     ...      ...    ...  ..       ...       ...         ...
29826  상일2동   95~99세      계 NaN        16        17         NaN
29827  상일2동   95~99세    한국인 NaN        16        17         NaN
29828  상일2동   95~99세  등록외국인 NaN         -         -         NaN
29829  상일2동  100세 이상      계 NaN         3         3         NaN
29830  상일2동  100세 이상    한국인 NaN         3         3         NaN

[29831 rows x 7 columns]


In [ ]:
# '동별'로 그룹화하고 총 그룹의 갯수 확인
total_dong_groups = len(df.groupby('동별'))

# 결과 출력
print("총 그룹화된 동의 갯수:", total_dong_groups)

# 여기엔 생각해보니까 .. 구랑 이상한 합계가 들어가는구나 지워야겠네

총 그룹화된 동의 갯수: 451


In [ ]:
# 필요한 행 선택
selected_rows = df[df['연령별'].isin(['15~19세', '20~24세', '25~29세', '30~34세', '35~39세'])]

# 선택한 행 출력
print(selected_rows)

# 새로운 데이터프레임을 CSV 파일로 저장
selected_rows.to_csv('/content/selected_data.csv', index=False, encoding='utf-8-sig')

print(len(selected_rows.groupby('동별')))

         동별     연령별     항목  단위 2023. 3/4 2023. 4/4  Unnamed: 6
12       합계  15~19세      계 NaN    390134    388426         NaN
13       합계  15~19세    한국인 NaN    382203    378197         NaN
14       합계  15~19세  등록외국인 NaN      7931     10229         NaN
15       합계  20~24세      계 NaN    607352    601307         NaN
16       합계  20~24세    한국인 NaN    563540    555058         NaN
...     ...     ...    ...  ..       ...       ...         ...
29788  상일2동  30~34세    한국인 NaN       853       848         NaN
29789  상일2동  30~34세  등록외국인 NaN         5         4         NaN
29790  상일2동  35~39세      계 NaN       822       815         NaN
29791  상일2동  35~39세    한국인 NaN       820       811         NaN
29792  상일2동  35~39세  등록외국인 NaN         2         4         NaN

[6780 rows x 7 columns]
451


In [ ]:
selected_df = selected_rows[selected_rows['동별'].str.endswith('동')]

print(selected_df)
print(len(selected_df.groupby('동별')))

# 잘 지운거 같은데

         동별     연령별     항목  단위 2023. 3/4 2023. 4/4  Unnamed: 6
144     사직동  15~19세      계 NaN       356       357         NaN
145     사직동  15~19세    한국인 NaN       352       352         NaN
146     사직동  15~19세  등록외국인 NaN         4         5         NaN
147     사직동  20~24세      계 NaN       420       425         NaN
148     사직동  20~24세    한국인 NaN       410       407         NaN
...     ...     ...    ...  ..       ...       ...         ...
29788  상일2동  30~34세    한국인 NaN       853       848         NaN
29789  상일2동  30~34세  등록외국인 NaN         5         4         NaN
29790  상일2동  35~39세      계 NaN       822       815         NaN
29791  상일2동  35~39세    한국인 NaN       820       811         NaN
29792  상일2동  35~39세  등록외국인 NaN         2         4         NaN

[6390 rows x 7 columns]
425


In [ ]:
real = selected_df[selected_df['항목'].str.endswith('계')]

print(real)
print(len(real.groupby('동별')))

         동별     연령별 항목  단위 2023. 3/4 2023. 4/4  Unnamed: 6
144     사직동  15~19세  계 NaN       356       357         NaN
147     사직동  20~24세  계 NaN       420       425         NaN
150     사직동  25~29세  계 NaN       666       660         NaN
153     사직동  30~34세  계 NaN       811       818         NaN
156     사직동  35~39세  계 NaN       685       656         NaN
...     ...     ... ..  ..       ...       ...         ...
29778  상일2동  15~19세  계 NaN       650       658         NaN
29781  상일2동  20~24세  계 NaN       759       734         NaN
29784  상일2동  25~29세  계 NaN       814       809         NaN
29787  상일2동  30~34세  계 NaN       858       852         NaN
29790  상일2동  35~39세  계 NaN       822       815         NaN

[2130 rows x 7 columns]
425


In [ ]:
real_2 = real.drop(['2023. 3/4', 'Unnamed: 6', '단위', '항목'], axis=1)

print(real_2)
print(len(real_2.groupby('동별')))

         동별     연령별 2023. 4/4
144     사직동  15~19세       357
147     사직동  20~24세       425
150     사직동  25~29세       660
153     사직동  30~34세       818
156     사직동  35~39세       656
...     ...     ...       ...
29778  상일2동  15~19세       658
29781  상일2동  20~24세       734
29784  상일2동  25~29세       809
29787  상일2동  30~34세       852
29790  상일2동  35~39세       815

[2130 rows x 3 columns]
425


In [ ]:
# '동별'로 그룹화하고 15세부터 39세까지의 열들을 선택하여 합계 계산
result_df = real_2.groupby('동별')['2023. 4/4'].sum().reset_index()

# 결과 출력
print(result_df)
print(len(result_df.groupby('동별')))

       동별             2023. 4/4
0    가락1동  11481124145922362598
1    가락2동  16181720232824162251
2    가락본동   9011301222724452102
3    가리봉동     15647613351149825
4     가산동   3771753551845842457
..    ...                   ...
420   효창동      3626969441078939
421   후암동    604807147316021261
422  휘경1동    638310125581301968
423  휘경2동  11233308325419101237
424   흑석동  12843309295121952068

[425 rows x 2 columns]
425


In [ ]:
# '2023. 4/4' 열의 데이터 형식 확인
print(real_2['2023. 4/4'].dtype)

# 만약 데이터 형식이 숫자가 아니라면 숫자로 변환
real_2['2023. 4/4'] = pd.to_numeric(real_2['2023. 4/4'], errors='coerce')

# 다시 '동별'로 그룹화하고 15세부터 39세까지의 열들을 선택하여 합계 계산
result_df = real_2.groupby('동별')['2023. 4/4'].sum().reset_index()

# 결과 출력
print(result_df)

result_df.to_csv('/content/각동별인구수정제.csv',index=False, encoding='utf-8-sig')

float64
       동별  2023. 4/4
0    가락1동     8565.0
1    가락2동    10333.0
2    가락본동     8976.0
3    가리봉동     3941.0
4     가산동    14689.0
..    ...        ...
420   효창동     4019.0
421   후암동     5747.0
422  휘경1동     8566.0
423  휘경2동    10832.0
424   흑석동    11807.0

[425 rows x 2 columns]


# 여기까지가 일단 각 동별 인구수 만들기 끝이고,
# 견인 현황 뒤에다가 동별 인구수 붙이기

In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import matplotlib.font_manager as fm

In [ ]:
df_tow=pd.read_csv('./서울특별시_전동킥보드_견인_현황_20230831.csv', encoding='cp949')
df_tow.shape

(123727, 6)

In [ ]:
unique_types=df_tow['유형'].unique()
print(unique_types)

['보도와 차도가 구분된 도로의 차도' '보도 중앙' '버스정류장, 택시 승강장' ...
 '[전화신고] [주민민원신고]동작구 상도로60길 38 그린빌라 벽쪽 주차공간에 현재 씽씽이 킥보드 4대 있음 즉시 수거바람'
 '구청 민원 미 조치\n와룡' '빨리 가져가세요']


In [ ]:
value_counts=df_tow['유형'].value_counts()
print(value_counts)

top_20_types=df_tow['유형'].value_counts().head(20)
print(top_20_types)

보도와 차도가 구분된 도로의 차도       58416
횡단보도, 산책로 등              17246
버스정류장, 택시 승강장            10086
보도 중앙                     9821
자전거 도로                    7166
                         ...  
차도전동킥보드불편신고                  1
횡단보도 주변 무단주차 다니기불편합니다        1
점자블록치워주세요                    1
공원관리자                        1
빨리 가져가세요                     1
Name: 유형, Length: 1469, dtype: int64
보도와 차도가 구분된 도로의 차도           58416
횡단보도, 산책로 등                  17246
버스정류장, 택시 승강장                10086
보도 중앙                         9821
자전거 도로                        7166
점자블록, 엘리베이터 입구                5530
지하철역 진출입로 및 주변                4177
기타                            4177
통행불편                           682
건물, 상가 보행자 진출입                 583
불편신고                           561
오신고                            265
차도                             201
횡단보도                           173
차도 사고위험                        167
차도 불법주차                        138
보행불편                           104
자전거도로                      

In [ ]:
retain_values=[
    '보도와 차도가 구분된 도로의 차도','횡단보도, 산책로 등','버스정류장, 택시승강장','보도 중앙','자전거 도로','점자블록, 엘리베이터 입구','지하철역 진출입로 및 주변','건물, 상가 보행자 진출입','차도','횡단보도','차도 사고위험','차도 불법주차','자전거도로']
df_tow=df_tow[df_tow['유형'].isin(retain_values)]
df_tow

번호         신고일   구정보                      주소                  유형  \
0            1  2021-07-15  영등포구   서울특별시 영등포구 여의도동 84-10  보도와 차도가 구분된 도로의 차도   
1            2  2021-07-15  영등포구  서울특별시 영등포구 영등포동1가 97-4               보도 중앙   
2            3  2021-07-15   동작구    서울특별시 동작구 노량진동 231-2  보도와 차도가 구분된 도로의 차도   
4            5  2021-07-15   동작구   서울특별시 동작구 신대방동 460-30  보도와 차도가 구분된 도로의 차도   
6            7  2021-07-15  영등포구     서울특별시 영등포구 신길동 4209         횡단보도, 산책로 등   
...        ...         ...   ...                     ...                 ...   
123657  123658  2023-08-31   송파구     서울특별시 송파구 문정동 641-1               자전거도로   
123671  123672  2023-08-31    중구   서울특별시 중구 신당동 432-1625             차도 불법주차   
123676  123677  2023-08-31  영등포구    서울특별시 영등포구 대림동 736-2                  차도   
123693  123694  2023-08-31    중구    서울특별시 중구 남대문로4가 45-2             차도 불법주차   
123706  123707  2023-08-31   동작구     서울특별시 동작구 상도동 535-5                  차도   

               조치일  
0       2021-07-15  
1       2021-07-15  
2       2021-07-15  
4       2021-07-15  
6       2021-07-15  
...            ...  
123657  2023-08-31  
123671  2023-08-31  
123676  2023-08-31  
123693  2023-08-31  
123706  2023-08-31  

[103717 rows x 6 columns]

In [ ]:
unique_types=df_tow['유형'].unique()
print(unique_types)

['보도와 차도가 구분된 도로의 차도' '보도 중앙' '횡단보도, 산책로 등' '건물, 상가 보행자 진출입'
 '점자블록, 엘리베이터 입구' '자전거 도로' '지하철역 진출입로 및 주변' '차도 사고위험' '횡단보도' '차도'
 '차도 불법주차' '자전거도로']


In [ ]:
df_tow_type=df_tow
df_tow_type['유형']=df_tow_type['유형'].replace('자전거도로', '자전거 도로')
df_tow_type['유형']=df_tow_type['유형'].replace('차도 사고위험', '차도')
df_tow_type['유형']=df_tow_type['유형'].replace('차도 불법주차', '차도')
df_tow_type['유형']=df_tow_type['유형'].replace('횡단보도, 산책로 등', '횡단보도')
df_tow_type['유형']=df_tow_type['유형'].replace('점자블록, 엘리베이터 입구', '엘리베이터 입구')
df_tow_type['유형']=df_tow_type['유형'].replace('건물, 상가 보행자 진출입', '진출입로')
df_tow_type['유형']=df_tow_type['유형'].replace('지하철역 진출입로 및 주변', '진출입로')

df_tow_type=df_tow_type['유형'].value_counts().reset_index()
df_tow_type.columns=['유형', '횟수']

df_tow_type

유형     횟수
0  보도와 차도가 구분된 도로의 차도  58416
1                횡단보도  17419
2               보도 중앙   9821
3              자전거 도로   7265
4            엘리베이터 입구   5530
5                진출입로   4760
6                  차도    506

In [ ]:
!sudo apt-get install -y fonts-nanum
!sudo fc-cache -fv
!rm ~/.cache/matplotlib -rf

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
fonts-nanum is already the newest version (20200506-1).
0 upgraded, 0 newly installed, 0 to remove and 35 not upgraded.
/usr/share/fonts: caching, new cache contents: 0 fonts, 1 dirs
/usr/share/fonts/truetype: caching, new cache contents: 0 fonts, 3 dirs
/usr/share/fonts/truetype/humor-sans: caching, new cache contents: 1 fonts, 0 dirs
/usr/share/fonts/truetype/liberation: caching, new cache contents: 16 fonts, 0 dirs
/usr/share/fonts/truetype/nanum: caching, new cache contents: 12 fonts, 0 dirs
/usr/local/share/fonts: caching, new cache contents: 0 fonts, 0 dirs
/root/.local/share/fonts: skipping, no such directory
/root/.fonts: skipping, no such directory
/usr/share/fonts/truetype: skipping, looped directory detected
/usr/share/fonts/truetype/humor-sans: skipping, looped directory detected
/usr/share/fonts/truetype/liberation: skipping, looped directory detected
/usr/share/fonts/truetype/

In [ ]:
duplicated_addresses=df_tow['주소'].duplicated()
print(duplicated_addresses)

0         False
1         False
2         False
4         False
6         False
          ...  
123657     True
123671     True
123676     True
123693     True
123706     True
Name: 주소, Length: 103717, dtype: bool


In [ ]:
print(duplicated_addresses.sum())

77921


In [ ]:
df_tow_location=df_tow['주소'].value_counts().reset_index()
df_tow_location.columns=['주소', '횟수']

df_tow_location

주소   횟수
0        서울특별시 마포구 동교동 192-44  762
1           서울특별시 마포구 서교동 415  678
2          서울특별시 서초구 서초동 1390  377
3      서울특별시 성동구 성수동2가 275-24  355
4          서울특별시 서초구 서초동 1327  329
...                       ...  ...
25791   서울특별시 동작구 사당동 708-642    1
25792   서울특별시 강서구 내발산동 659-10    1
25793  서울특별시 서대문구 홍제동 287-226    1
25794  서울특별시 성동구 하왕십리동 1066-2    1
25795    서울특별시 중랑구 망우동 442-24    1

[25796 rows x 2 columns]

# 바로 밑에 2개 실행하지 마세요 !!!!!!!! 저 밑으로 내려가시오 !!!!

In [ ]:
import re

# 각 행에 대해 '~동' 추출하는 함수
def extract_dong_name(address):
    match = re.search(r'(\w+동)', address)
    return match.group(1) if match else None

# df_tow_location 데이터프레임에 새로운 열 추가
df_tow_location['동'] = df_tow_location['주소'].apply(extract_dong_name)

# 결과 확인
print(df_tow_location[['주소', '동']])


                           주소     동
0        서울특별시 마포구 동교동 192-44   동교동
1           서울특별시 마포구 서교동 415   서교동
2          서울특별시 서초구 서초동 1390   서초동
3      서울특별시 성동구 성수동2가 275-24    성동
4          서울특별시 서초구 서초동 1327   서초동
...                       ...   ...
25791   서울특별시 동작구 사당동 708-642   사당동
25792   서울특별시 강서구 내발산동 659-10  내발산동
25793  서울특별시 서대문구 홍제동 287-226   홍제동
25794  서울특별시 성동구 하왕십리동 1066-2    성동
25795    서울특별시 중랑구 망우동 442-24   망우동

[25796 rows x 2 columns]


In [ ]:
# '/content/각동별인구수정제.csv' 파일 불러오기
population_df = pd.read_csv('/content/각동별인구수정제.csv', encoding='utf-8-sig')

# '주소' 열에서 동 이름 추출하여 새로운 열 '동이름' 생성
df_tow_location['동이름'] = df_tow_location['주소'].str.extract(r'(\S+동)')

# '동이름' 열을 기준으로 '/content/각동별인구수정제.csv'에서 인구수를 가져와서 '인구수' 열 생성
df_tow_location['인구수'] = df_tow_location['동이름'].map(population_df.set_index('동별')['2023. 4/4'])

# '동이름' 열 삭제
df_tow_location.drop('동이름', axis=1, inplace=True)

# 결과 확인
print(df_tow_location)


                           주소   횟수     동      인구수
0        서울특별시 마포구 동교동 192-44  762   동교동      NaN
1           서울특별시 마포구 서교동 415  678   서교동  13872.0
2          서울특별시 서초구 서초동 1390  377   서초동      NaN
3      서울특별시 성동구 성수동2가 275-24  355    성동      NaN
4          서울특별시 서초구 서초동 1327  329   서초동      NaN
...                       ...  ...   ...      ...
25791   서울특별시 동작구 사당동 708-642    1   사당동      NaN
25792   서울특별시 강서구 내발산동 659-10    1  내발산동      NaN
25793  서울특별시 서대문구 홍제동 287-226    1   홍제동      NaN
25794  서울특별시 성동구 하왕십리동 1066-2    1    성동      NaN
25795    서울특별시 중랑구 망우동 442-24    1   망우동      NaN

[25796 rows x 4 columns]


뭐지 왜 안되는거지 ... 를 확인해본 결과 ... 레전드 사실 발견 !!

바로 .... 법정동과 행정동이 다르다는 사실... 보통 통계는 행정동에 따라서 인구수를 통계내기에 주소에 나와 있는 동은 법정동임 .... 그래서 바로 위에 0번을 보면 주소에는 '동교동'이라고 적혀있는데, 사실 그게 행정동에는 존재하지 않는 동임 .... 레전드

그럼 어떡하지 ..? 카카오맵 api를 통해서 25795개의 데이터를 다 집어넣어서 .. 각각의 주소에 해당하는 행정동을 반환해주기는 하던데 ... 해주는게 1개당 1초 꼴인것 같았음 ... 즉 1분은 60초, 1시간은 3600초, 즉 2만 5천개하려면 대충 10시간이 걸린다는 사실 ..?

일단 바로 밑에 코드는 25000개 다 확인하는 코든데 ... 내가 10분 정도 돌려본 결과 진짜 10시간 걸리는거 맞는 거 같음 1개당 1초임..

고로 지금 자야하니까 수면 분석 켜놓고 잘거니까 .. 일단 우리가 중요하다고 생각한 52개에 대해서 먼저 해보겠음 ...

In [ ]:
import requests

# 발급받은 카카오맵 API 키
kakao_api_key = '838ed4e900821a97b1a6244b29ff14e1'

def get_dong_from_address(address):
    try:
        # 주소로부터 행정동 정보를 조회하는 API 요청
        url = f'https://dapi.kakao.com/v2/local/search/address.json'
        headers = {'Authorization': f'KakaoAK {kakao_api_key}'}
        params = {'query': address}
        response = requests.get(url, headers=headers, params=params)
        data = response.json()

        # API 응답에서 행정동 정보 추출
        dong = data['documents'][0]['address']['region_3depth_h_name']

        return dong
    except Exception as e:
        print(f'Error fetching data for address {address}: {str(e)}')
        return None

# 주소에 대해 행정동 정보 가져오기
df_tow_location['행정동'] = df_tow_location['주소'].apply(get_dong_from_address)

# 결과 확인
print(df_tow_location[['주소', '행정동']])


Error fetching data for address 서울특별시 송파구 문정동 374: list index out of range
Error fetching data for address 서울특별시 강서구 마곡동 769-4: list index out of range
Error fetching data for address 서울특별시 송파구 문정동 340: list index out of range
Error fetching data for address 서울특별시 도봉구 방학동 726-72: list index out of range
Error fetching data for address 서울특별시 송파구 문정동 341: list index out of range
Error fetching data for address 서울특별시 송파구 문정동 218: list index out of range
Error fetching data for address 서울특별시 송파구 문정동 323-1: list index out of range
Error fetching data for address 서울특별시 송파구 문정동 325-6: list index out of range


KeyboardInterrupt: 

여기서부턴 많이 중복된 애들에 대한 코드임 !!

In [ ]:
df_tow_location['횟수']=df_tow_location['횟수'].astype(int)

In [ ]:
df_tow_location=df_tow_location[df_tow_location['횟수']>30]
df_tow_location

주소   횟수
0      서울특별시 마포구 동교동 192-44  762
1         서울특별시 마포구 서교동 415  678
2        서울특별시 서초구 서초동 1390  377
3    서울특별시 성동구 성수동2가 275-24  355
4        서울특별시 서초구 서초동 1327  329
..                      ...  ...
425   서울특별시 영등포구 당산동5가 14-4   31
426    서울특별시 성동구 사근동 102-17   31
427   서울특별시 영등포구 대림동 1064-3   31
428    서울특별시 영등포구 여의도동 8-13   31
429      서울특별시 동작구 사당동 1112   31

[430 rows x 2 columns]

In [ ]:
import requests

# 발급받은 카카오맵 API 키
kakao_api_key = '838ed4e900821a97b1a6244b29ff14e1'

def get_dong_from_address(address):
    try:
        # 주소로부터 행정동 정보를 조회하는 API 요청
        url = f'https://dapi.kakao.com/v2/local/search/address.json'
        headers = {'Authorization': f'KakaoAK {kakao_api_key}'}
        params = {'query': address}
        response = requests.get(url, headers=headers, params=params)
        data = response.json()

        # API 응답에서 행정동 정보 추출
        dong = data['documents'][0]['address']['region_3depth_h_name']

        return dong
    except Exception as e:
        print(f'Error fetching data for address {address}: {str(e)}')
        return None

# 주소에 대해 행정동 정보 가져오기
df_tow_location['행정동'] = df_tow_location['주소'].apply(get_dong_from_address)

# 결과 확인
print(df_tow_location)


Error fetching data for address 서울특별시 송파구 문정동 374: list index out of range
Error fetching data for address 서울특별시 강서구 마곡동 769-4: list index out of range
Error fetching data for address 서울특별시 송파구 문정동 340: list index out of range
     index                      주소   횟수         위도          경도     행정동
0        0    서울특별시 마포구 동교동 192-44  762  37.557142  126.926190     서교동
1        1       서울특별시 마포구 서교동 415  678  37.555071  126.923825     서교동
2        2      서울특별시 서초구 서초동 1390  377  37.488180  127.017510    서초4동
3        3  서울특별시 성동구 성수동2가 275-24  355  37.540480  127.056010  성수2가3동
4        4      서울특별시 서초구 서초동 1327  329  37.488180  127.017510    서초2동
..     ...                     ...  ...        ...         ...     ...
348    424     서울특별시 관악구 봉천동 975-3   31  37.484665  126.948496     중앙동
349    425   서울특별시 영등포구 당산동5가 14-4   31  37.532080  126.901020    당산2동
350    426    서울특별시 성동구 사근동 102-17   31  37.558995  127.048028     사근동
351    427   서울특별시 영등포구 대림동 1064-3   31  37.495757  126.899936 

In [ ]:
# '/content/각동별인구수정제.csv' 파일 불러오기
population_df = pd.read_csv('/content/각동별인구수정제.csv', encoding='utf-8-sig')

# '행정동' 열을 기준으로 '/content/각동별인구수정제.csv'에서 인구수를 가져와서 '인구수' 열 생성
df_tow_location['인구수'] = df_tow_location['행정동'].map(population_df.set_index('동별')['2023. 4/4'])

# 결과 확인
print(df_tow_location)


NameError: name 'pd' is not defined

NameError: name 'df_tow_location' is not defined

In [ ]:
df_tow_location.to_csv('주소_횟수_행정동_좌표_인구수.csv',index=False, encoding='utf-8-sig')

이야호 ...... 430개는 6분동안 돌아가네 나이스 ....

In [ ]:
address=df_tow_location['주소']
print(address.head())

0      서울특별시 마포구 동교동 192-44
1         서울특별시 마포구 서교동 415
2        서울특별시 서초구 서초동 1390
3    서울특별시 성동구 성수동2가 275-24
4        서울특별시 서초구 서초동 1327
Name: 주소, dtype: object


In [ ]:
for i in range(len(address)):
    a = address[i].split(' ')
    address[i] = " ".join(a[0:4])
print(address)

0        서울특별시 마포구 동교동 192-44
1           서울특별시 마포구 서교동 415
2          서울특별시 서초구 서초동 1390
3      서울특별시 성동구 성수동2가 275-24
4          서울특별시 서초구 서초동 1327
                ...          
425     서울특별시 영등포구 당산동5가 14-4
426      서울특별시 성동구 사근동 102-17
427     서울특별시 영등포구 대림동 1064-3
428      서울특별시 영등포구 여의도동 8-13
429        서울특별시 동작구 사당동 1112
Name: 주소, Length: 430, dtype: object


<ipython-input-97-2edbd2da39d0>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  address[i] = " ".join(a[0:4])


In [ ]:
pip install geopy

In [ ]:
from geopy.geocoders import Nominatim
geo_local = Nominatim(user_agent='South Korea')

def geocoding(address):
    try:
        geo = geo_local.geocode(address)
        x_y = [geo.latitude, geo.longitude]
        return x_y
    except:
        return [0,0]

In [ ]:
latitude = []
longitude =[]

for i in address:
    latitude.append(geocoding(i)[0])
    longitude.append(geocoding(i)[1])

In [ ]:
df_tow_location = pd.DataFrame({'주소': df_tow_location['주소'],'횟수':df_tow_location['횟수'], '위도':latitude,'경도':longitude})

df_tow_location.to_csv('전처리 견인 데이터셋 위경도.csv')

In [ ]:
df_tow_location

주소   횟수         위도          경도
0      서울특별시 마포구 동교동 192-44  762  37.557142  126.926190
1         서울특별시 마포구 서교동 415  678  37.555071  126.923825
2        서울특별시 서초구 서초동 1390  377  37.488180  127.017510
3    서울특별시 성동구 성수동2가 275-24  355  37.540480  127.056010
4        서울특별시 서초구 서초동 1327  329  37.488180  127.017510
..                      ...  ...        ...         ...
425   서울특별시 영등포구 당산동5가 14-4   31  37.532080  126.901020
426    서울특별시 성동구 사근동 102-17   31  37.558995  127.048028
427   서울특별시 영등포구 대림동 1064-3   31  37.495757  126.899936
428    서울특별시 영등포구 여의도동 8-13   31   0.000000    0.000000
429      서울특별시 동작구 사당동 1112   31  37.485696  126.972352

[430 rows x 4 columns]

In [ ]:
df_tow_location = df_tow_location[(df_tow_location['위도'] != 0.000000) & (df_tow_location['경도'] != 0.000000)]
df_tow_location

주소   횟수         위도          경도
0      서울특별시 마포구 동교동 192-44  762  37.557142  126.926190
1         서울특별시 마포구 서교동 415  678  37.555071  126.923825
2        서울특별시 서초구 서초동 1390  377  37.488180  127.017510
3    서울특별시 성동구 성수동2가 275-24  355  37.540480  127.056010
4        서울특별시 서초구 서초동 1327  329  37.488180  127.017510
..                      ...  ...        ...         ...
424     서울특별시 관악구 봉천동 975-3   31  37.484665  126.948496
425   서울특별시 영등포구 당산동5가 14-4   31  37.532080  126.901020
426    서울특별시 성동구 사근동 102-17   31  37.558995  127.048028
427   서울특별시 영등포구 대림동 1064-3   31  37.495757  126.899936
429      서울특별시 동작구 사당동 1112   31  37.485696  126.972352

[353 rows x 4 columns]

In [ ]:
df_tow_location = df_tow_location.reset_index()
df_tow_location

index                      주소   횟수         위도          경도
0        0    서울특별시 마포구 동교동 192-44  762  37.557142  126.926190
1        1       서울특별시 마포구 서교동 415  678  37.555071  126.923825
2        2      서울특별시 서초구 서초동 1390  377  37.488180  127.017510
3        3  서울특별시 성동구 성수동2가 275-24  355  37.540480  127.056010
4        4      서울특별시 서초구 서초동 1327  329  37.488180  127.017510
..     ...                     ...  ...        ...         ...
348    424     서울특별시 관악구 봉천동 975-3   31  37.484665  126.948496
349    425   서울특별시 영등포구 당산동5가 14-4   31  37.532080  126.901020
350    426    서울특별시 성동구 사근동 102-17   31  37.558995  127.048028
351    427   서울특별시 영등포구 대림동 1064-3   31  37.495757  126.899936
352    429      서울특별시 동작구 사당동 1112   31  37.485696  126.972352

[353 rows x 5 columns]

In [ ]:
df_tow_location['경도'] = df_tow_location['경도'].astype(float)
df_tow_location['위도'] = df_tow_location['위도'].astype(float)

In [ ]:
import folium

m = folium.Map(location=[df_tow_location['위도'].iloc[0], df_tow_location['경도'].iloc[0]], zoom_start=11)

for _, row in df_tow_location.iterrows():
    folium.Circle(
        location=[row['위도'], row['경도']],
        radius=100,
        color='blue',
        fill=True,
        fill_color='cyan'
    ).add_to(m)

m